src/jit/lower.cpp

```diff
@@ -3803,7 +3803,11 @@ GenTree* Lowering::LowerVirtualStubCall(GenTreeCall* call)
             // on x64 we must materialize the target using specific registers.
             addr->gtRegNum = comp->virtualStubParamInfo->GetReg();
 
+// On ARM we must use a proper address in R12(thunk register) without dereferencing.
+// So for the jump we use the default register.
+#ifndef _TARGET_ARM_
             indir->gtRegNum = REG_JUMP_THUNK_PARAM;
```

Comment:
Does the register need to be ever set for any platform? Should this rather be deleted?

Reply:
I'll try it on tests.